saga/sagaState.go

```diff
@@ -5,28 +5,28 @@ import (
 	"fmt"
 )
 
+type flag byte
```

Comment:
SortOf...  You may not see an End message if CompMessages are supplied.

These are all valid combos of messages
(Start)
(Start, End)
(Start, StartComp)
(Start, End, StartComp)
(Start, StartComp, EndComp)
(Start, End, StartComp, EndComp)

Given that I can't get the flags to be smaller than 8 bits (I'm only using 4), I think its worth the clarity of having a flag for each received message.